bundles/org.openhab.binding.shelly/doc/images/AdvancedUsers.md

```diff
@@ -0,0 +1,94 @@
+You specify the device's IP and device model SHSW-25 and the page will generate you the link for the firmware download using the OTA of the device.
+Then you run "curl -s [-u user:password] &gt;generated link&gt;" from the terminal.
+This should show a JSON result, make sure that it shows "status:updating".
+Wait 15sec and access the device's Web UI, go to Settings:Firmware Upgrade and make sure than the new version was installed successful.
+- Manual download and installation of the firmware
+Manually pick the download link from the [Shelly Firmware Repository](https://api.shelly.cloud/files/firmware) and get the release or beta link.
+Once you downloaded the file you need to copy it to an http server. 
+Open the following url http://&lt;shelly ip&gt;/ota?url=http://&lt;web server&gt;/&lt;path&gt;/&lt;zip-file&gt;
+Again, make sure that the file is downloaded and installed properly.
+
+## Trouble Shooting
+
+### Network Settings
+
+Shelly devices do only support IPv4. 
+This implies that the openHAB host system has IPv4 bound to the network interface.
+The binding is only able to discover devices on the local subnet. 
+Add things manually with the given IP if you have a routed network in between or using a VPN connection.
+
+The binding enables CoIoT protocol by default if the device is running firmware 1.6 or newer.
+CoIoT is based on CoAP and uses a UDP based signaling using IP Multicast (224.0.1.187, port 5683).
+Again if the device is not on the same local IP subnet you need special router/switch configurations to utilized CoAP via IP Multicast.
+Otherwise disable the Auto-CoIoT feature in the binding config (not the thing config), disable CoIoT events in the thing configuration and enable sensors events (http callback).
+Nevertheless in this setup the binding can communicate the device, but you are loosing the benefits of CoIoT.
+
+Refer to openHAB's general documentation when running openHAB in a docker container. Enabling mDNS discovery has additional setup requirements.  
+
+### Re-discover when IP address has changed
+ 
+Important: The IP address should not be changed after the device is added to openHAB.
+
+This can be achieved by
+
+- assigning a static IP address (recommended for battery powered devices) or
+- using DHCP and setup the router to always assign the same IP address to the device
+
+When the IP address changes for a device you need to delete the Thing and then re-discover the device.
+In this case channel linkage gets lost and you need to re-link the channels/items.
+
+## Log optimization
+
+The binding provides channels (e.g. heartBeat, currentWatts), which might cause a log of log output, especially when having multiple dozen Shellys.
+
+openHAB has an integrated feature to filter the event log.
+This mechanism doesn't filter the event, but the log output (items still receive the updates).
+
+A configuration is added as a new section to `openhab2-userdata/etc/org.ops4j.pax.logging.cfg`
```

Comment:
The directory name has changed to `openhab-userdata` in OH3. Also, the default logging file is now log4j2.xml, which is XML based.

Reply:
hmm, could you provide me the matching OH3 Info, currently I'm not really in OH3, otherwhise I need to remove this section (which was driven by users, because updates to currentWatt and other rapid changing channels could flood the log)

Reply:
Here is the new config file: https://github.com/openhab/openhab-distro/blob/master/distributions/openhab/src/main/resources/userdata/etc/log4j2.xml

Reply:
ok, that looks similar to the old format still used in the IDE setup
but how could I define filters specifically removing those frequent messages from the logfile, e.g. based on item names to text patterns. adapting the 2.5.x definition:
```
# custom filtering rules
log4j2.appender.event.filter.uselessevents.type = RegexFilter
log4j2.appender.event.filter.uselessevents.regex = .*(heartBeat|LastUpdate|lastUpdate|LetzteAktualisierung|Uptime|Laufzeit|ZuletztGesehen).*
log4j2.appender.event.filter.uselessevents.onMatch = DENY
log4j2.appender.event.filter.uselessevents.onMisMatch = NEUTRAL
```

Reply:
I would need to dive into it, too. Although if we cobble something together we'd need to test it.

Reply:
@fwolter I removed the section for now. I hadn't the time to dig into details.I'll include this in the upcoming PR